src/OpenFeature.Contrib.Providers.Flagd/FlagdConfig.cs

```diff
@@ -50,6 +71,18 @@ internal int MaxEventStreamRetries
             set { _maxEventStreamRetries = value; }
         }
 
+        internal ResolverType ResolverType
+        {
+            get => _resolverType;
+            set => _resolverType = value;
+        }
+
+        internal string SourceSelector
+        {
+            get { return _sourceSelector; }
+            set { _sourceSelector = value; }
+        }
+
```

Comment:
In Kubernmetes, the ENV_VAR based configs make a lot of sense, but I think we need a way to use the in-process provider in code as well, so users who aren't un K8s can use whatever configuration tools they want to configure flagd directly through code.

I think we should either allow configuration through `FlagdConfig` (which would mean exposing a public constructor for it, with public fields) or add another constructor on the provider that takes a URL and Resolver type. Whatever is easier, I think (though I should point out we also have a public `FlagdProvider` constructor which takes a `FlagdConfig` which is sort-of useless now I think.

Reply:
+1 for exposing a configuration-based constructor. This should be easily done through a builder. And I can point to Java [1] & go flagd [2] providers which allows both env vars and constructor paramters 

[1] - https://github.com/open-feature/java-sdk-contrib/tree/main/providers/flagd#in-process-resolver 
[2] - https://github.com/open-feature/go-sdk-contrib/tree/main/providers/flagd#in-process-resolver